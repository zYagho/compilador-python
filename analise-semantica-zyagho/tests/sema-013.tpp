{Erro: Chamada para a função 'principal' não permitida}
{Aviso: Atribuição de tipos distintos. Coerção implícita do valor atribuído '10' do tipo 'inteiro' para 'a' que é 'flutuante'}
{Aviso: Atribuição de tipos distintos. Coerção implícita do valor retornado por 'func' é do tipo 'inteiro' para 'a' que é 'flutuante'}
{Aviso: Variável 'b' declarada e inicializada, mas não utilizada}
{Erro: Função 'principal' deveria retornar 'inteiro', mas retorna 'vazio'}
{Erro: Função 'func' deveria retornar 'inteiro', mas retorna 'flutuante'}

flutuante: a
inteiro: b

inteiro func()
  a := 10
  principal()
  retorna(a)
fim

inteiro principal()
	b := 18
	a := func()
fim
