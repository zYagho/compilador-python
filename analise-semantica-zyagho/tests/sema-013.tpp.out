ERR-SEM-CALL-FUNC-MAIN-NOT-ALLOWED
WAR-SEM-IMP-COERC-OF-NUM
WAR-SEM-VAR-DECL-INIT-NOT-USED
WAR-SEM-IMP-COERC-OF-RET-VAL
ERR-SEM-FUNC-RET-TYPE-ERROR
ERR-SEM-FUNC-RET-TYPE-ERROR
